src/java.xml.crypto/share/classes/org/jcp/xml/dsig/internal/dom/Policy.java

```diff
@@ -44,13 +44,13 @@
 public final class Policy {
 
     // all restrictions are initialized to be unconstrained
-    private static Set<URI> disallowedAlgs = new HashSet<>();
-    private static int maxTrans = Integer.MAX_VALUE;
-    private static int maxRefs = Integer.MAX_VALUE;
-    private static Set<String> disallowedRefUriSchemes = new HashSet<>();
-    private static Map<String, Integer> minKeyMap = new HashMap<>();
-    private static boolean noDuplicateIds = false;
-    private static boolean noRMLoops = false;
+    private static Set<URI> disallowedAlgs;
+    private static int maxTrans;
+    private static int maxRefs;
+    private static Set<String> disallowedRefUriSchemes;
+    private static Map<String, Integer> minKeyMap;
+    private static boolean noDuplicateIds;
+    private static boolean noRMLoops;
```

Comment:
Does it really make any difference, moving the initialization to the initialize() method? The comment on line 46 no longer apply now that the initialization is relocated.

Reply:
No for good people. I was writing some local tests to add and remove rules in an arbitrary style, and cleaning up inside the `initialize()` method makes this possible, albeit still a little ugly (need `--add-opens`).

I'll remove that comment.